pypit/core/arextract.py

```diff
@@ -30,6 +30,934 @@
+    fit a simple Gaussian. Port of IDL LOWREDUX long_gprofile.pro
+
+     Parameters
+     ----------
+     image : numpy float 2-d array [nspec, nspat]
+         sky-subtracted image
+     ivar : numpy float 2-d array [nspec, nspat]
+         inverse variance of sky-subtracted image
+     waveimg numpy float 2-d array [nspec, nspat]
+         2-d wavelength map
+     trace_in : numpy 1-d array [nspec]
+         object trace
+     wave : numpy 1-d array [nspec]
+         extracted wavelength of spectrum
+     flux : numpy 1-d array [nspec]
+         extracted flux of spectrum
+     fluxivar : numpy 1-d array [nspec]
+         inverse variance of extracted flux spectrum
+
+
+    Optional Parameters
+    ----------
+    thisfwhm : float
+         fwhm of the object trace
+    MAX_TRACE_CORR : float [default = 2.0]
+         maximum trace correction to apply
+    SN_GAUSS : float [default = 3.0]
+         S/N ratio below which code just uses a Gaussian
+    wvmnx : float [default = [2900.0,30000.0]
+         wavelength range of usable part of spectrum
+    hwidth : float [default = None]
+         object maskwidth determined from object finding algorithm. If = None,
+         code defaults to use 3.0*(np.max(thisfwhm) + 1.0)
+    PROF_NSIGMA : float [default = None]
+         Number of sigma to include in the profile fitting. This option is only needed for bright objects that are not
+         point sources, which allows the profile fitting to fit the high S/N wings (rather than the default behavior
+         which truncates exponentially). This allows for extracting all the flux and results in better sky-subtraction
+         for bright extended objects.
+    NO_DERIV : boolean [default = False]
+         disables determination of derivatives and exponential tr
+
+     Returns
+     -------
+     :func:`tuple`
+         A tuple containing the (sset, outmask, yfit, reduced_chi), where
+
+            sset: object
+               bspline object
+            outmask: : :class:`numpy.ndarray`
+               output mask which the same size as xdata
+            yfit  : :class:`numpy.ndarray`
+               result of the bspline fit (same size as xdata)
+            reduced_chi: float
+               value of the reduced chi^2
+     """
+
+    from scipy.interpolate import interp1d
+    if hwidth is None: 3.0*(np.max(thisfwhm) + 1.0)
+    if PROF_NSIGMA is not None:
+        NO_DERIV = True
+
+    thisfwhm = np.fmax(thisfwhm,1.0) # require the FWHM to be greater than 1 pixel
+
+    xnew = trace_in
+
+    nspat = image.shape[1]
+    nspec = image.shape[0]
+
+    # create some images we will need
+    profile_model = np.zeros((nspec,nspat))
+    sub_obj = image
+    sub_ivar = ivar
+    sub_wave = waveimg
+    sub_trace = trace_in
+    sub_x = np.arange(nspat)
+    sn2_sub = np.zeros((nspec,nspat))
+    spline_sub = np.zeros((nspec,nspat))
+
+
+
+    flux_sm = djs_median(flux, width = 5, boundary = 'reflect')
```

Comment:
I didn't see djs_median get imported.

Should be:

```
from pypit.core import pydl
```

then
```
pydl.djs_median
```

Although use Numpy if you can.